Remove finput and fgrad_input from slow3d signatures

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69044
* #68899
* #68898
* __->__ #68897

Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

jbschlosser committed:
Remove finput and fgrad_input from slow3d signatures

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).

zou3519 commented on aten/src/ATen/native/ConvolutionMM3d.cpp:
```diff
@@ -343,12 +395,27 @@ void slow_conv3d_backward_out_cpu_template(
+      (input_width + 2 * pad_width - kernel_width) / stride_width + 1;
   const int64_t batch_size = input.size(0);
 
+  Tensor fgrad_input = at::empty({batch_size,
+      n_input_plane * kernel_depth * kernel_height * kernel_width,
+      output_depth * output_height * output_width}, input.options());
+  TORCH_CHECK(fgrad_input.is_contiguous(), "fgrad_input must be contiguous")
```
this TORCH_CHECK doesn't seem necessary? since we're constructing a tensor with at::empty it should come out contiguous anyways, right?

jbschlosser replied:
good catch & I agree!

zou3519 commented on aten/src/ATen/native/ConvolutionMM3d.cpp:
```diff
@@ -15,6 +15,86 @@ namespace native {
+      auto columns_a = columns.accessor<scalar_t, 3>();
+
+      at::parallel_for(0, batch_size, CONV3D_GRAIN_SALT, [&](int64_t start, int64_t end) {
+        for (int64_t t = start; t < end; t++) {
```
nit: `for (const auto t : c10::irange(start, end)) {` ?

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).

jbschlosser committed:
Update on "Remove finput and fgrad_input from slow3d signatures"


Differential Revision: [D32655875](https://our.internmc.facebook.com/intern/diff/D32655875)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655875).
